lib/wsdl.js

```diff
@@ -1932,10 +1932,22 @@ WSDL.prototype.findSchemaType = function(name, nsURI) {
   return schema.complexTypes[name];
 };
 
-WSDL.prototype.findChildSchemaObject = function(parameterTypeObj, childName) {
+WSDL.prototype.findChildSchemaObject = function(parameterTypeObj, childName, backtrace) {
   if (!parameterTypeObj || !childName) {
     return null;
   }
+
+  if (!backtrace) {
+    backtrace = [];
+  }
+
+  if (backtrace.indexOf(parameterTypeObj) >= 0) {
+    // We've recursed back to ourselves; break.
+    return null;
+  } else {
+    backtrace = backtrace.concat([parameterTypeObj]);
```

Comment:
just to understand it better: what's the purpose of using `concat()` and creating a new `Array` instead of just `push()`ing ?